src/ssh/cockpitsshtransport.c

```diff
@@ -720,13 +701,13 @@ cockpit_ssh_transport_class_init (CockpitSshTransportClass *klass)
          g_param_spec_boolean ("prompt-hostkey", NULL, NULL, FALSE,
                                G_PARAM_WRITABLE | G_PARAM_CONSTRUCT_ONLY | G_PARAM_STATIC_STRINGS));
 
-  g_object_class_install_property (object_class, PROP_CREDS,
-         g_param_spec_boxed ("creds", NULL, NULL, COCKPIT_TYPE_CREDS,
-                             G_PARAM_WRITABLE | G_PARAM_CONSTRUCT_ONLY | G_PARAM_STATIC_STRINGS));
+  g_object_class_install_property (object_class, PROP_PASSWORD,
+         g_param_spec_string ("password", NULL, NULL, NULL,
+                              G_PARAM_READWRITE | G_PARAM_CONSTRUCT_ONLY | G_PARAM_STATIC_STRINGS));
```

Comment:
The fact that password is a string means it's copied throughout memory here. We should be using a g_param_spec_boxed of type G_TYPE_BYTES here. However, this could be a follow up.